checker/src/main/java/org/checkerframework/checker/nullness/CollectionToArrayHeuristics.java

```diff
@@ -28,33 +36,73 @@
  *   <li value="2">The passed array size is less than the collection size. Here are heuristics to
  *       handle the most common cases:
  *       <ol>
- *         <li value="1">an empty array initializer, e.g. {@code c.toArray(new String[] {})},
- *         <li value="2">array creation tree of size 0, e.g. {@code c.toArray(new String[0])}, or
- *         <li value="3">array creation tree of the collection size method invocation {@code
- *             c.toArray(new String[c.size()])}
+ *         <li value="1">the argument has length 0:
+ *             <ol>
+ *               <li value="1">an empty array initializer, e.g. {@code c.toArray(new String[] {})},
+ *                   or
+ *               <li value="2">array creation tree of size 0, e.g. {@code c.toArray(new String[0])}.
+ *             </ol>
+ *         <li value="2">array creation tree with a collection {@code size()} method invocation as
+ *             argument {@code c.toArray(new String[c.size()])}
  *       </ol>
  * </ol>
  *
- * Note: The nullness of the returned array doesn't depend on the passed array nullness.
+ * <p>Additionally, when the lint option {@link NullnessChecker#LINT_TRUSTARRAYLENZERO} is provided,
+ * a call to {@link Collection#toArray(Object[]) Collection.toArray(T[])} will be estimated to
+ * return an array with a non-null component type if the argument is a field access where the field
+ * declaration has a {@code @ArrayLen(0)} annotation. This trusts the {@code @ArrayLen(0)}
+ * annotation, but does not verify it. Run the Constant Value Checker to verify that annotation.
+ *
+ * <p>Note: The nullness of the returned array doesn't depend on the passed array nullness. This is
+ * a fact about {@link Collection#toArray(Object[]) Collection.toArray(T[])}, not a limitation of
+ * this heuristic.
+ *
+ * @checker_framework.manual #nullness-collection-toarray Nullness and conversions from collections
+ *     to arrays
+ * @checker_framework.manual #constant-value-checker Constant Value Checker
  */
+// Note: The https://checkerframework.org/manual/#nullness-collection-toarray section in the manual,
```

Comment:
I checked the two texts, and they contained some subtle differences in wording.  I think this will be difficult to maintain.  I suggest that you remove the documentation from here and instead just reference the manual.